### ISSUE TITLE
TestCase inheritance breaks with setUpTestData (Django 1.8)

### ISSUE THREAD

**Issue by bforchhammer** [pytest-dev/pytest-django#265]
Hi,

We have a large test base with test cases which inherit from each other. I am in the process of converting them to Django 1.8 and tried to use the new `setUpTestData` class method for setting up test data.

This fails in a very strange way for tests using inheritance: `setUpTestData` is only invoked for the base class and not for the derived class. I think this may be because of [this commit](https://github.com/pytest-dev/pytest-django/commit/8a9aacbd835a49064ff8999137fa6ede34f3c520), which changes and overrides `setUpClass` and `tearDownClass` in a way which I honestly don't quite understand...

I have created a [failing test](https://github.com/pytest-dev/pytest-django/commit/107ba5b51aabc087d7c69078043c8c3816278ae1) which shows the problem; the lines which cause errors are annotated with `TODO` comments.

Any ideas on how to fix or circumvent this problem are greatly appreciated! :-)

**Comment by blueyed** [comment]
Yes, it looks like the new instancemethods should use `super()` to call the parents..

I think the way forward would be to create a PR with your failing test (great!), and then try to fix it from there.

**Comment by pelme** [comment]
Thanks for the test case, that should make a fix for this much easier!

I'd like to have this fixed before the next release, I will try to take a stab at it the next couple of days if no one else wants to give it a try.

**Comment by pelme** [comment]
@bforchhammer I have a fix for this in PR #277 - could you give it a try and make sure it fixes your problem?

```
pip install -e git+https://github.com/pelme/pytest-django.git@multi-inheritance-testcase#egg=pytest-django
```

**Comment by pelme** [comment]
The fix is now merged in master, I'm closing this issue now. Please re-open if there are any problems with the fix!
